### DISCUSSION TITLE
Default permissions when creating new file

### DISCUSSION THREAD

**Discussion by dpetka2001** [nvim-neo-tree/neo-tree.nvim#708] [Q&A] [answered]
Hi I would like to know if there is any way to change the default permissions of newly created files in neo-tree.
Currently the newly created files have 644 permissions and I would like to change this to 664. Thanks in advance.

**Comment by pysan3** [comment, marked as answer]
Unfortunately the permissions are hard-coded [here](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8238865e1d9c61f1a260c290653f2c419503e0a9/lua/neo-tree/sources/filesystem/lib/fs_actions.lua#L355).

But you can write a `file_added` handler to overwrite the permission right after the files / directories are created. Something like this should work.

(This implementation does not handle permissions of parent directories created together (if any) tho. Feel free to customize it if you need.)

*Tested only on Linux...

```lua
event_handlers = {
  {
    event = "file_added",
    handler = function (destination)
      local uv = vim.loop
      local file_info = uv.fs_stat(destination)
      if not file_info then
        return
      elseif file_info.type == "file" then
        uv.fs_chmod(destination, 436) -- (436 base 10) == (664 base 8)
      elseif file_info.type == "directory" then
        uv.fs_chmod(destination, 609) -- 644 does not work for directories I guess?
      end
    end,
  }
}
```

**Reply by dpetka2001** [reply]
Thank you very much for your help. I will try it out and come back if I have any further problems.

**Reply by dpetka2001** [reply]
It worked perfectly for my needs. Appreciate the help very much!!
